src/main/java/talkabot/TalkaBot.java

```diff
@@ -0,0 +1,166 @@
+                    this.ui.echo(curr, this.taskList.size());
+                    storage.save(this.taskList);
+                } else if (input.toLowerCase().startsWith("event")) {
+                    if (input.length() < 7) {
+                        throw new InvalidScheduleException();
+                    }
+                    Task curr = new Event(Parser.getEvent(input));
+                    this.taskList.add(curr);
+                    this.ui.echo(curr, this.taskList.size());
+                    storage.save(this.taskList);
+                } else {
+                    if (input == "") {
+                    } else {
+                        throw new UnknownInputException(input);
+                    }
+                }
+            } catch (TalkaBotException e) {
+                this.ui.error(e.getMessage());
+            } catch (DateTimeException e) {
+                this.ui.error("Sorry, I need a valid date format! For example: yyyy-mm-dd");
+            } catch (IOException e) {
+                this.ui.error(e.getMessage());
+            }
+            if (!end) {
+                ui.dashedLine();
+            }
+        }
+        this.ui.goodbye();
+    }
+
+    private boolean isValidNumber(String str, int len) {
+        try {
+            return str.length() > len
+                    && Integer.parseInt(str.substring(len)) <= this.taskList.size()
+                    && Integer.parseInt(str.substring(len)) >= 1;
+        } catch(NumberFormatException e){
+            return false;
+        }
+    }
+
+    private void mark(String input) throws InvalidEditException, IOException {
+        if (!isValidNumber(input, 5)) {
+            throw new InvalidEditException("mark");
+        }
+        Task task = this.taskList.get(Integer.parseInt(input.substring(5)) - 1);
+        task.markAsDone();
+        this.ui.mark(task);
+        storage.save(this.taskList);
+    }
+
+    private void unmark(String input) throws InvalidEditException, IOException {
+        if (!isValidNumber(input, 7)) {
+            throw new InvalidEditException("unmark");
+        }
+        Task task = this.taskList.get(Integer.parseInt(input.substring(7)) - 1);
+        task.markAsUndone();
+        this.ui.unmark(task);
+        storage.save(this.taskList);
+    }
+
+    private void delete(String input) throws InvalidEditException, IOException {
+        if (!isValidNumber(input, 7)) {
+            throw new InvalidEditException("delete");
+        }
+        Task task = this.taskList.delete(Integer.parseInt(input.substring(7)) - 1);
+        this.ui.delete(task, this.taskList.size());
+        storage.save(this.taskList);
+    }
+
+    private void getDay(String input) throws InvalidEditException {
+        if (!isValidNumber(input, 8)) {
+            throw new InvalidEditException("get the day of");
+        }
+        Task task = this.taskList.get(Integer.parseInt(input.substring(8)) - 1);
+        this.ui.getDay(task);
+    }

```

Comment:
Ahh I see that these are private methods so you did not include Javadoc comments for them. Although not necessary, I thought it would be good if you include some Javadoc comments for these as well so that if you reference to them next time, you will know what the method does easily. Just a thought!